packages/@aws-cdk/aws-pipes-alpha/lib/pipe.ts

```diff
@@ -233,6 +233,10 @@ export class Pipe extends PipeBase {
      */
     const source = props.source.bind(this);
     props.source.grantRead(this.pipeRole);
+    if (props.source instanceof SourceWithDlq) {
```

Comment:
`instanceof` doesn't always play nice with jsii. I can't remember the specifics of why off the top of my head, but I'd avoid using it here.

Reply:
What's the preferred route here?

Reply:
A static guard method should work well:

```ts
class SourceWithDlq implements ISource {
  //...
  static isSourceWithDlq(source: ISource): source is SourceWithDlq {
    return (source as SourceWithDlq).sourceArn !== undefined;
  }
}